### DISCUSSION TITLE
How to use TIME ZONE AT?

### DISCUSSION THREAD

**Discussion by Janjanus** [linq2db/linq2db#4688] [Q&A] [answered]
I am trying to translate a query that uses `AT TIME ZONE` to linq2db. In Entity Framework I can use the: `EF.Functions.AtTimeZone(date, timeZoneName)` that maps to the correct SQL. 

How can I do this in linq2db?

I have tried using `EF.Functions.AtTimeZone(date, timeZoneName)` but this doesn't work correctly.

**Comment by MaceWindu** [comment, marked as answer]
E.g.
```cs
[Sql.Expression("{0} AT TIME ZONE {1}", IsNullable = Sql.IsNullableType.SameAsFirstParameter)]
public static int? AtTimeZone(DateTimeOffset? datetime, string tzId) => throw new InvalidOperationException();
```
